### PR TITLE
[CONTINT-3908] Add tests for dogstatsd origin detection on k8s

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds tests for dogstatsd origin detection using udp on k8s.

### CODE REVIEW

File: test/new-e2e/tests/containers/k8s_test.go [hunk 1]
```diff
@@ -599,60 +601,64 @@ func (suite *k8sSuite) TestDogstatsdStandalone() {
...
 func (suite *k8sSuite) testDogstatsd(kubeNamespace string) {
-	// Test dogstatsd origin detection with UDS
+	suite.testDogstatsdContainerID(kubeNamespace, kubeDeploymentDogstatsdUDS)
+	suite.testDogstatsdContainerID(kubeNamespace, kubeDeploymentDogstatsdUDPOrigin)
+	// Test dogstatsd origin detection with UDP
 	suite.testMetric(&testMetricArgs{
 		Filter: testMetricFilterArgs{
 			Name: "custom.metric",
 			Tags: []string{
-				"kube_deployment:dogstatsd-uds",
```
Comment (AliDatadog) [inline test/new-e2e/tests/containers/k8s_test.go:607]: The git diff shows my changes in a strange way.

**Review by wdhif — APPROVED** [review]

**Pull request merged**